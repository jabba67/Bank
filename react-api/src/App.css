.App {
  text-align: center;
  
}

.Container{
  background-attachment: scroll;  
  padding-right: 50px;
  padding-left: 50px;
  padding-top: 50px;
  padding-bottom: 50px;
  justify-content: center;
  backdrop-filter: blur(5px);
  background-color: rgba(255, 255, 255, .15);
  max-width: 300vh;
  position: absolute;
  top: 50%;
  left: 50%;
  margin-right: -50%;
  transform: translate(-50%, -50%);
  white-space: pre-line;
}

.SignIn{
  max-width: 300vh;
  padding-right: 50px;
  padding-left: 50px;
  position: absolute;
  top: 50%;
  left: 50%;
  margin-right: -50%;
  transform: translate(-50%, -50%);
  backdrop-filter: blur(5px);
  background-color: rgba(255, 255, 255, .15);  
}
.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  background-color:white;
  min-height: 100vh;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color:black;
  background-image: url(https://wallpaperaccess.com/full/2005304.jpg);
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}
