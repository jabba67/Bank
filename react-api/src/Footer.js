import React from 'react';
import {Container} from 'react-bootstrap';
import Row from 'react-bootstrap/Row';
import {Col} from 'react-bootstrap';

function Footer(){
    return(
    <div className="Footer">
        <footer className = "mt-16 flex flex-col justify-center items-center">
            <Container fluid = {true}>
               <Row className = "border-top justify-content-between p-2">
                  <Col className = "p-0" md={3} sm={12}>
                    Tyler Rubin
                  </Col> 
                  <Col className ="p-0 d-flex justify-content-end" md={3}>
                    This site was made by Tyler Rubin
                  </Col> 
               </Row> 
            </Container>
        </footer>
    </div>
    );
}
export default Footer;