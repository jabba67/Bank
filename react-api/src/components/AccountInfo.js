import React, { Component } from "react";
import {Card,CardHeader,CardTitle,CardImg,CardBody,CardFooter} from "shards-react";
import EmailAddress from './grabEmail';
import CheckingAccountNumbers from "./grabCheckingAccountNumber";
import AccountNumbers from "./grabAccountNumber";
import App from '../App'
import Content from '../components/Content';
const axios = require('axios'); 

export default class AccountInfo extends React.Component{

  state = {
    accountnumber: [],
    checkingaccountnumber: []
  }

componentWillMount() {
    fetch(`https://localhost:44358/api/UserInformations/${this.props.userEmail}`)
    .then(res => res.json())
    .then((data) =>  {
      this.setState({ accountnumber: data })
    })
    .catch(console.log)

    fetch(`https://localhost:44358/api/UserInformations/${this.props.userEmail}`)
  .then(res => res.json())
  .then((data) =>  {
    this.setState({checkingaccountnumber: data})
  })
  .catch(console.log)

}
  render() {
    return (
      <div class = "AccountInfo">
        <center>
        <Card style={{maxHeight:"360px", maxWidth: "370px", backgroundColor: 'white'}}>
                        <CardHeader>Account Info</CardHeader>
                        <CardBody>
                          <CardTitle>      ALL ACCOUNT INFO      </CardTitle>
                          <p>Account Number:  <AccountNumbers accountnumber ={this.state.accountnumber}/></p>
                          <p>Checking Account Number:  <CheckingAccountNumbers checkingaccountnumber ={this.state.checkingaccountnumber}/></p>
                        </CardBody>
                      </Card>
        </center>
      </div>
    );
  }
}