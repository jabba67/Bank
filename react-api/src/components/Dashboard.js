import React, { Component } from "react";
import ReactDOM from 'react-dom';
import Container from 'react-bootstrap/Container';
import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';
import {Card} from 'react-bootstrap';
import { Avatar } from 'antd';
import { UserOutlined } from '@ant-design/icons';
import { Dialog, DialogActionsBar } from '@progress/kendo-react-dialogs';
import { Input } from '@progress/kendo-react-inputs'; 
import { Button } from '@progress/kendo-react-buttons';
import { savePDF } from '@progress/kendo-react-pdf';
import { Ripple } from '@progress/kendo-react-ripple';

//Import CSS
import 'bootstrap-4-grid/css/grid.min.css';
import '@progress/kendo-theme-material/dist/all.css';
import '../App.css';

//Import Components and Assets
import CheckingAccountNumbers from "./grabCheckingAccountNumber";
import AccountNumbers from "./grabAccountNumber";
import { DonutChartContainer } from './DonutChartContainer';
import { BarChartContainer } from './BarChartContainer';
import { GridContainer } from './GridContainer';
import { PanelBarContainer } from './PanelBarContainer';
import moneySign from '../moneySign.png'
import bankCover from '../bankCover.jpg'
const axios = require('axios'); 

export default class Dashboard extends React.Component{

  constructor(props) {
    super(props);
    this.appContainer = React.createRef();
    /*this.state = {
      showDialog: false
    }*/
  }

  //State Declaration
  state = {
    accountnumber: [],
    checkingaccountnumber: []
  }

  //Button Functions
  handlePDFExport = () => {
    savePDF(ReactDOM.findDOMNode(this.appContainer), { paperSize: 'auto' });
  }

  handleShare = () => {
    this.setState({
      showDialog: !this.state.showDialog
    }, () => console.log(this.state))
  }

  componentWillMount() {
      fetch(`https://localhost:44358/api/UserInformations/${this.props.userEmail}`)
      .then(res => res.json())
      .then((data) =>  {
        this.setState({ accountnumber: data })
      })
      .catch(console.log)

      fetch(`https://localhost:44358/api/UserInformations/${this.props.userEmail}`)
    .then(res => res.json())
    .then((data) =>  {
      this.setState({checkingaccountnumber: data})
    })
    .catch(console.log)
  }

  render() {
    return (
    <div className="app-container container" ref={(el) => this.appContainer = el}>
      <Ripple>
        <div className="bootstrap-wrapper">
          <div className="app-container container">
            <div className="row">
              <div className="col-xs-6 col-sm-6 col-md-6 col-lg-6 col-xl-6">
                <h1>Welcome $Name Here$</h1>
              </div>
              <div className="col-xs-6 col-sm-6 col-md-6 col-lg-6 col-xl-3">
                <Button primary={true} onClick={this.handleShare}>Share</Button>
                <Button onClick={this.handlePDFExport}>Export to PDF</Button>
              </div>
            </div>
            <div className="row">
              {/*<div className="col-xs-3 col-sm-3 col-md-3 col-lg-3 col-xl-3">
                <PanelBarContainer/>
    </div>*/}
              
                <div className="row">
                  <div className="col-xs-6 col-sm-6 col-md-4 col-lg-4 col-xl-4">
                    <h4>Spending Summary</h4>
                    <DonutChartContainer />
                  </div>
                  <div className="col-xs-6 col-sm-6 col-md-5 col-lg-2 col-xl-2">
                    <h4>Investment Summary</h4>
                    <div className="percentage-container">
                      <span className="percentage-number">94</span>
                      <span className="percentage-sign">%</span>
                      <p>Total Return</p>
                    </div>
                    <div className="percentage-container">
                      <span className="percentage-number">30</span>
                      <span className="percentage-sign">%</span>
                      <p>Target Return</p>
                    </div>
                  </div>
                  <div className="col-xs-12 col-sm-12 col-md-6 col-lg-6 col-xl-6">
                    <h4>Average Balance</h4>
                    <BarChartContainer/>
                  </div>
                </div>
                <div className="row">
                  <div className="col-xs-12 col-sm-12 col-md-12 col-lg-12 col-xl-12">
                    <h4>Transactions</h4>
                    <GridContainer />
                  </div>
                 
                  </div>
            </div>
            {this.state.showDialog &&
              <Dialog title={"Share this report"} onClose={this.handleShare}>
                <p>Please enter the email address/es of the recipient/s.</p>
                <Input placeholder="example@progress.com" />
                <DialogActionsBar>
                  <Button primary={true} onClick={this.handleShare}>Share</Button>
                  <Button onClick={this.handleShare}>Cancel</Button>
                </DialogActionsBar>
              </Dialog>
            }
          </div>
        </div>
      </Ripple>
      </div>
    );
  }
}