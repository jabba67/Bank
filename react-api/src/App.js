import React, { Component, useState, useEffect, Image, ImageBackground, useReducer } from 'react';
import {Route,HashRouter, BrowserRouter as Router} from "react-router-dom";
import ReactDOM from 'react-dom';
import withFirebaseAuth from 'react-with-firebase-auth'
import * as firebase from 'firebase/app';
import 'firebase/auth';
import firebaseconfig from './firebaseconfig';
import './App.css';
import {Alert,Breadcrumb,BreadcrumbItem,Container,Row,Col,Button,
  ButtonGroup,ButtonToolbar,Form,FormGroup,FormInput,InputGroup,
  Card,CardHeader,CardTitle,CardImg,CardBody,CardFooter,Navbar,
  NavbarToggler,NavbarBrand,Nav,NavItem,NavLink,Dropdown,DropdownToggle,
  DropdownMenu,DropdownItem,InputGroupAddon,Collapse,
  InputGroupText} from "shards-react";
import Contacts from './components/contacts';
import AccountNumbers from './components/grabAccountNumber';
import TransHistory from './components/grabTransactionHistory';
import AccountBalance from './components/AccountBalance';
import AccountInfo from './components/AccountInfo';

const firebaseApp = firebase.initializeApp(firebaseconfig);

///Console testing for API data
const element = fetch("https://localhost:44358/api/UserInformations/45505")
      .then((response) => response.json())
      .then((data) => console.log('Here is my data returned:', data.accountNumber));
///Console testing

const axios = require('axios');

class App extends React.Component{
  constructor() {
    super();
    this.handleSubmit = this.handleSubmit.bind(this);
    this.input = React.createRef();
    this.handlelogIn = this.handlelogIn.bind(this);
    this.handleChange = this.handleChange.bind(this);
  }

  state = {
    contacts: [],
    datas: [],
    transHistory: [],
    account: " "
  }

  componentDidMount() {
    console.log("happening in comp1");
    /*fetch(`https://localhost:44358/api/UserInformations/${this.state.account}`)
    .then(res => res.json())
    .then((data) =>  {
      this.setState({ contacts: data })
    })
    .catch(console.log)

    fetch(`https://localhost:44358/api/UserInformations/${this.state.account}`)
    .then(res => res.json())
    .then((data) =>  {
      this.setState({ datas: data })
    })
    .catch(console.log)

    fetch(`https://localhost:44358/api/TransactionTrackings/${this.state.account}`)
    .then(res => res.json())
    .then((data) =>  {
      this.setState({ transHistory: data })
    })
    .catch(console.log)*/
  }//End componentDidMount()

  handlelogIn(event) {
    event.preventDefault();
    console.log(this.state.acount)
    axios.get(`https://localhost:44358/api/UserInformations/${this.state.account}`)
    .then(response => {
      console.log(response.data)
      this.setState({ datas: response.data})
      this.setState({ contacts: response.data})
      console.log(this.state.datas)
    })
    .then(
      this.props.signInWithGoogle()
    )
  }

  handleChange(event){
    this.setState({ account: event.target.value  })
    console.log(this.state.account)
  }

  handleSubmit(event) {
    event.preventDefault();
    alert('A name was submitted: ' + this.input.current.value);
    var accountBalance;
    var current = this.state.contacts.accountBalance;
    console.log('the value of contacts rn is: ' + this.state.contacts.accountBalance);
    accountBalance = (parseInt(this.input.current.value)+ this.state.contacts.accountBalance);
    
    /*axios({
      method: 'POST',
      url: 'https://localhost:44358/api/UserInformations/45505',
      data: {
        firstName: "Tyler",
        accountBalance: accountBalance,
        lastName: "Rubin",
        age: 25,
        birthDate: "04/15/1995",
        socialSecurityNumber: "533-14-1324",
        address: "1242 Tallow Tree Lane",
        phoneNumber: "858-342-0865",
        emailAddress: "arcowirexzs@yahoo.com",
        accountNumber: 45505,
        password: "jabba678"
      }
    });*/

    axios({
      method: 'PATCH',
      url: 'https://localhost:44358/api/UserInformations/45505',
      data: {
        accountBalance: accountBalance,
        accountNumber: 45505
      }
    });
      //IMPLEMENT THIS: Send request to API to get account number based on validation email
      //Then With the account number perform actions like GET and PATCH on the stored account number
  }
  
  render() {
    const {
      user,
      signOut,
      signInWithGoogle,
    } = this.props;
    return (
      <div className="App" >
        <header className="App-header">
          {
            user
              ? <>
              <Router>
                <Nav className="justify-content-center">
                <NavItem></NavItem>
                <NavItem></NavItem>
                  <NavItem>
                    <li><NavLink href="/">Home</NavLink></li>
                    <Route path="/App" component={App}/> 
                    <li><NavLink href="/AccountBalance">Account Balance</NavLink></li>
                    <Route path="/AccountBalance" component={AccountBalance}/>
                    <li><NavLink href="/AccountInfo" >Account Info</NavLink></li>
                    <Route path="/AccountInfo" component={AccountInfo} email = {user.email} /> 
                    <AccountInfo userName = {user.email}/> 
                  </NavItem>
                  <NavItem>
                    <NavLink onClick={signOut}>Sign Out</NavLink>
                  </NavItem>
                  <NavItem></NavItem>
                  <NavItem>
                  </NavItem>
                </Nav><br></br><br></br>
              </Router>

              <div className="Container">
                {/*
                <p align = "center">Hello, {user.displayName}</p>
                <Container className="dr-example-container"style={{ backgroundColor: 'transparent'}} align = "center" maxHeight ="500px" maxWidth = "350px" >
                <Row>
                  <Col>
                  </Col>
                  
                  <Col></Col>
                </Row>
                <Row>
                    <Col></Col>
                    <Col sm="25" lg="30">
                  
                </Col>
                <Col></Col>
              </Row>
                </Container>*/}
              </div>

              </>
              : <div class = "SignIn">
                <br></br>
              <Card style={{maxHeight:"370", maxWidth: "380", backgroundColor: 'white'}}>
                  <CardHeader>Sign In</CardHeader>
                  <CardImg src="https://t3.ftcdn.net/jpg/02/20/14/38/240_F_220143804_fc4xRygvJ8bn8JPQumtHJieDN4ORNyjs.jpg" />
                  <CardBody>
                    <form onSubmit={this.handlelogIn}>
                      <label>Enter Account Number: 
                        <br></br>
                        <input type="text"  value={this.state.account} onChange={this.handleChange}/>
                        <FormInput placeholder="Account Number" input type="submit" value="log-in" />
                      </label>
                    </form>
                  </CardBody>
              </Card>
              <br></br>
              </div>
          }
          
        </header>
      </div>
    );//End of Return
  }//End of Render()
}//End Class App

const firebaseAppAuth = firebaseApp.auth();

const providers = {
  googleProvider: new firebase.auth.GoogleAuthProvider(),
};

export default withFirebaseAuth({
  providers,
  firebaseAppAuth,
})(App);

//<AccountInfo userName = {user.email}/>