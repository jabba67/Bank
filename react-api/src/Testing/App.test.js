/* eslint-disable no-undef */
/* eslint-disable linebreak-style */
import React, { Component } from 'react';
import Enzyme from "enzyme";
import { mount } from 'enzyme';
import { shallow } from 'enzyme';
import Adapter from "enzyme-adapter-react-16";
import renderer from 'react-test-renderer';
import App from '../App';
import AccountBalance from '../components/AccountBalance';
import Footer from '../Footer';
import SignUp from '../components/signUp';
import AccountInfo from '../components/AccountInfo';
import Dashboard from '../components/Dashboard';
import TransactionGrid from '../components/TransactionsGridContainer';

Enzyme.configure({adapter: new Adapter() });
//Use it.skip to skip a test when using Jest

describe("Basic App Tests", ()=>{
  it("App Renders without crashing", ()=>{
    const wrapper = shallow(<App/>)
    expect(wrapper).toMatchSnapshot();
  })
  
  it.skip("Renders Sign Up Label", () => {
    const wrapper = shallow(<App/>);
    const signupLabel = <a href="/signUp"> Sign Up Here!</a>;
    expect(wrapper.contains(signupLabel)).toEqual(true);
  });
})

describe("Main Component Snapshot Tests",() =>{
  test('App Snapshot test', () => {
    const component = renderer.create(<App/>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  
  test('Account Balance Snapshot test', () => {
    const component = renderer.create(<AccountBalance />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  
  test('Footer Snapshot test', () => {
    const component = renderer.create(<Footer />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  
  test('Signup Snapshot test', () => {
    const component = renderer.create(<SignUp />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  
  test('Account Info Snapshot test', () => {
    const component = renderer.create(<AccountInfo />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  
  /*test('Dashboard Snapshot test', () => {
    const component = renderer.create(<Dashboard />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  
  test('TransactionGrid Snapshot test', () => {
    const component = renderer.create(<TransactionGrid />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });*/
})




