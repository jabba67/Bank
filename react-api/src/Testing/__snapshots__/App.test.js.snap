// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Account Balance Snapshot test 1`] = `
<div
  className="app-container container"
>
  <div
    class="AccountBalance"
    style={
      Object {
        "backgroundColor": "#e8e8e8",
      }
    }
  >
    <section
      id="accountBalances"
    >
      <h4>
        ACCOUNT BALANCES
        <br />
        <img
          alt="Account Icon"
          height={50}
          src="account3.svg"
          width={50}
        />
      </h4>
      <div
        className="row"
      >
        <div
          className="col-xs-6 col-sm-6 col-md-6 col-lg-6 col-xl-6"
        >
          <img
            alt="Home Icon"
            height={25}
            src="home-outline.svg"
            width={50}
          />
          Main Account Balance:
          <div />
          <div />
        </div>
        <div
          className="col-xs-6 col-sm-6 col-md-6 col-lg-6 col-xl-6"
        >
          <img
            alt="Cash Icon"
            height={25}
            src="cash-outline.svg"
            width={50}
          />
          Checking Account Balance:
          <div />
          <div />
          <br />
        </div>
      </div>
    </section>
    <section
      id="accountActions"
    >
      <h4>
        ACCOUNT ACTIONS
        <br />
        <img
          alt="Transfer Icon"
          height={50}
          src="transfer.svg"
          width={50}
        />
      </h4>
      <div
        className="row"
      >
        <div
          className="col-xs-6 col-sm-6 col-md-6 col-lg-6 col-xl-6"
        >
          <h4>
            DEPOSIT
            <br />
            <img
              alt="Deposit Icon"
              height={50}
              src="deposit.svg"
              width={50}
            />
          </h4>
          <div
            align="center"
          >
            <form
              onSubmit={[Function]}
            >
              <label
                className="form-label"
              >
                Deposit to Main Account:
                <br />
                <input
                  placeholder="        Main Account"
                  type="text"
                />
              </label>
              <br />
              <br />
              <label>
                Deposit to Checking Account:
                <br />
                <input
                  placeholder="     Checking Account"
                  type="text"
                />
              </label>
              <br />
              <input
                onClick={[Function]}
                type="submit"
                value="Deposit"
              />
            </form>
          </div>
          <br />
        </div>
        <div
          className="col-xs-6 col-sm-6 col-md-6 col-lg-6 col-xl-6"
        >
          <h4>
            WIDTHDRAW
            <br />
            <img
              alt="Withdraw Icon"
              height={50}
              src="withdraw2.svg"
              width={50}
            />
          </h4>
          <div
            align="center"
          >
            <form
              onSubmit={[Function]}
            >
              <label
                className="form-label"
              >
                   Withdraw from Main Account:
                <br />
                <input
                  placeholder="        Main Account"
                  type="text"
                />
              </label>
              <br />
              <br />
              <label>
                 Withdraw from Checking Account:
                <br />
                <input
                  placeholder="     Checking Account"
                  type="text"
                />
              </label>
              <br />
              <input
                onClick={[Function]}
                type="submit"
                value="Withdraw"
              />
            </form>
          </div>
          <br />
        </div>
      </div>
    </section>
    <div
      className="row"
    >
      <div
        className="col-xs-12 col-sm-12 col-md-12 col-lg-12 col-xl-12"
      >
        <section
          id="accountBalances"
        >
          <center>
            <div
              className="card-title h5"
            >
              <h4>
                RECENT TRANSACTIONS:
              </h4>
            </div>
            <div>
               
              <section>
                <ul
                  className="list"
                >
                  <li
                    className="card"
                  >
                    <span>
                      <span
                        className="react-loading-skeleton css-1vmnjpn-skeletonStyles-Skeleton"
                        style={
                          Object {
                            "height": 180,
                          }
                        }
                      >
                        ‌
                      </span>
                    </span>
                    <h4
                      className="card-title"
                    >
                      <span>
                        <span
                          className="react-loading-skeleton css-1vmnjpn-skeletonStyles-Skeleton"
                          style={
                            Object {
                              "height": 36,
                              "width": "80%",
                            }
                          }
                        >
                          ‌
                        </span>
                      </span>
                    </h4>
                    <p
                      className="card-channel"
                    >
                      <span>
                        <span
                          className="react-loading-skeleton css-1vmnjpn-skeletonStyles-Skeleton"
                          style={
                            Object {
                              "width": "60%",
                            }
                          }
                        >
                          ‌
                        </span>
                      </span>
                    </p>
                    <div
                      className="card-metrics"
                    >
                      <span>
                        <span
                          className="react-loading-skeleton css-1vmnjpn-skeletonStyles-Skeleton"
                          style={
                            Object {
                              "width": "90%",
                            }
                          }
                        >
                          ‌
                        </span>
                      </span>
                    </div>
                  </li>
                </ul>
              </section>
            </div>
          </center>
        </section>
      </div>
    </div>
  </div>
</div>
`;

exports[`Account Info Snapshot test 1`] = `
<div
  className="AccountInfo"
>
  <div
    align="justified"
    className="container-fluid"
  >
    <div
      className="justify-content-center py-4 row"
    >
      <div
        className="col-md-3 col-sm-12"
      >
        <div
          className="card bg-transparent text-light border-transparent"
        >
          <img
            className="card-img-top"
            src="bankCover.jpg"
          />
          <div
            className="card-body"
          >
            <div
              className="card-title h5"
            >
               
              <span
                className="ant-avatar ant-avatar-circle"
                style={
                  Object {
                    "verticalAlign": "middle",
                  }
                }
              >
                <span
                  className="ant-avatar-string"
                  style={
                    Object {
                      "opacity": 0,
                    }
                  }
                >
                  <img
                    height={30}
                    src="moneySign.png"
                    width={30}
                  />
                </span>
              </span>
               
              ALL ACCOUNT INFO
            </div>
            <p>
              Account Number:
              <div />
            </p>
            <p>
              Checking Account Number:
              <div />
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`App Renders without crashing 1`] = `ShallowWrapper {}`;

exports[`App Snapshot test 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <div
      className="SignIn"
    >
      <div
        id="tsparticles"
      >
        <canvas
          className="example"
          style={
            Object {
              "height": 768,
              "width": 1024,
            }
          }
        />
      </div>
      <div
        className="card border-white"
        style={
          Object {
            "backgroundColor": "white",
            "borderBottomLeftRadius": 0,
            "borderBottomRightRadius": 0,
          }
        }
      >
        <center>
          <img
            alt="Titan Bank Logo"
            height={310}
            src="TitanBankDraft3.png"
            width={745}
          />
        </center>
        <div
          className="card-body"
        >
          <span
            className="animate-on-change animate-on-change-"
            onAnimationEnd={[Function]}
            onTransitionEnd={[Function]}
            style={
              Object {
                "animationDuration": "200ms",
                "display": "inline-block",
                "opacity": 1,
                "transition": "opacity 200ms ease-out",
              }
            }
          >
            <img
              alt="Sign in With Google button"
              height={62}
              onClick={[Function]}
              src="googleLogin.png"
              width={257}
            />
          </span>
          <br />
          <br />
          <label>
             Don't Have An Account? 
          </label>
          <a
            href="/signUp"
          >
             Sign Up Here!
          </a>
        </div>
      </div>
      <div
        className="card-columns"
        style={
          Object {
            "backgroundColor": "white",
            "border": "white",
            "borderBottomLeftRadius": 5,
            "borderBottomRightRadius": 5,
          }
        }
      >
        <div
          className="card border-white"
          style={
            Object {
              "width": "18rem",
            }
          }
        >
          <img
            alt="Account Management Icon"
            height={50}
            src="account4.svg"
            width={50}
          />
          <div
            className="card-body"
          >
            <div
              className="card-title h5"
            >
              Account Management
            </div>
            <p
              className="card-text"
            >
              All accounts are displayed clearly in one place.
            </p>
          </div>
        </div>
        <div
          className="card border-white"
          style={
            Object {
              "width": "18rem",
            }
          }
        >
          <img
            alt="Savings Management Icon"
            height={50}
            src="savings.svg"
            width={50}
          />
          <div
            className="card-body"
          >
            <div
              className="card-title h5"
            >
              Savings Management
            </div>
            <p
              className="card-text"
            >
              Saving money has never been easier with our modern implementation of saving methods!
            </p>
          </div>
        </div>
        <div
          className="card border-white"
          style={
            Object {
              "width": "18rem",
            }
          }
        >
          <img
            alt="Card Management Icon"
            height={50}
            src="card3.svg"
            width={50}
          />
          <div
            className="card-body"
          >
            <div
              className="card-title h5"
            >
              Card Management
            </div>
            <p
              className="card-text"
            >
              Users will never lose track of their cards balance.
            </p>
          </div>
        </div>
        <div
          className="card border-white"
          style={
            Object {
              "width": "18rem",
            }
          }
        >
          <img
            alt="Loans Management Icon"
            height={50}
            src="loans.svg"
            width={50}
          />
          <div
            className="card-body"
          >
            <div
              className="card-title h5"
            >
              Loans
            </div>
            <p
              className="card-text"
            >
              We enable our users to track loans and apply for new ones instantly.
            </p>
          </div>
        </div>
        <div
          className="card border-white"
          style={
            Object {
              "width": "18rem",
            }
          }
        >
          <img
            alt="Contracts Management Icon"
            height={50}
            src="contract.svg"
            width={50}
          />
          <div
            className="card-body"
          >
            <div
              className="card-title h5"
            >
              Contracts
            </div>
            <p
              className="card-text"
            >
              All contracts signed between the user and the bank feature online signatures to let the user avoid unnecessary visits to the bank.
            </p>
          </div>
        </div>
        <div
          className="card border-white"
          style={
            Object {
              "width": "18rem",
            }
          }
        >
          <img
            alt="Investment Management Icon"
            height={50}
            src="investment.svg"
            width={50}
          />
          <div
            className="card-body"
          >
            <div
              className="card-title h5"
            >
              Investment Mangement
            </div>
            <p
              className="card-text"
            >
              Our bank provides investment insights for clients to see the performance and make decisions based on frequently updated information.
            </p>
          </div>
        </div>
      </div>
    </div>
  </header>
</div>
`;

exports[`Footer Snapshot test 1`] = `
<div
  className="Footer"
>
  <section
    id="footer"
  >
    <footer
      className="mt-16 flex flex-col justify-center items-center"
    >
      <div
        className="container-fluid"
      >
        <br />
        <br />
        <br />
        This site was made by Tyler Rubin
        <br />
        <center>
          LinkedIn:
        </center>
        <center>
          Github:
        </center>
        <br />
        <br />
        <br />
      </div>
    </footer>
  </section>
</div>
`;

exports[`Signup Snapshot test 1`] = `
<div>
  <br />
  <div
    bg="transparent"
    className="container-fluid"
  >
    <form
      className=""
      onSubmit={[Function]}
    >
      <div
        className="form-group row"
      >
        <label
          className="form-label"
          htmlFor="formPlaintextEmail"
        >
          First Name
        </label>
        <input
          className="form-control"
          id="formPlaintextEmail"
          name="name"
          onChange={[Function]}
          placeholder="First Name"
          type="text"
          value=""
        />
      </div>
      <div
        className="form-group row"
      >
        <label
          className="form-label"
          htmlFor="formPlaintextEmail"
        >
          Last Name
        </label>
        <input
          className="form-control"
          id="formPlaintextEmail"
          name="name"
          onChange={[Function]}
          placeholder="Last Name"
          type="text"
          value=""
        />
      </div>
      <div
        className="form-group row"
      >
        <label
          className="form-label"
          htmlFor="formPlaintextEmail"
        >
          Email 
        </label>
        <input
          className="form-control"
          id="formPlaintextEmail"
          name="name"
          onChange={[Function]}
          placeholder="Email"
          type="text"
          value=""
        />
      </div>
      <div
        className="form-group row"
      >
        <label
          className="form-label"
          htmlFor="formPlaintextEmail"
        >
          Password
        </label>
        <input
          className="form-control"
          id="formPlaintextEmail"
          name="name"
          onChange={[Function]}
          placeholder="Password"
          type="text"
          value=""
        />
      </div>
      <div
        className="form-group row"
      >
        <label
          className="form-label"
          htmlFor="formPlaintextEmail"
        >
          Age
        </label>
        <input
          className="form-control"
          id="formPlaintextEmail"
          name="name"
          onChange={[Function]}
          placeholder="Age"
          type="text"
          value={0}
        />
      </div>
      <div
        className="form-group row"
      >
        <label
          className="form-label"
          htmlFor="formPlaintextEmail"
        >
          Birth Date
        </label>
        <input
          className="form-control"
          id="formPlaintextEmail"
          name="name"
          onChange={[Function]}
          placeholder="DD/MM/YYYY"
          type="text"
          value=""
        />
      </div>
      <div
        className="form-group row"
      >
        <label
          className="form-label"
          htmlFor="formPlaintextEmail"
        >
          Social Security: 
        </label>
        <input
          className="form-control"
          id="formPlaintextEmail"
          name="name"
          onChange={[Function]}
          placeholder="Social Security"
          type="text"
          value=""
        />
      </div>
      <div
        className="form-group row"
      >
        <label
          className="form-label"
          htmlFor="formPlaintextEmail"
        >
          Address
        </label>
        <input
          className="form-control"
          id="formPlaintextEmail"
          name="name"
          onChange={[Function]}
          placeholder="Address"
          type="text"
          value=""
        />
      </div>
      <div
        className="form-group row"
      >
        <label
          className="form-label"
          htmlFor="formPlaintextEmail"
        >
          Phone Number
        </label>
        <input
          className="form-control"
          id="formPlaintextEmail"
          name="name"
          onChange={[Function]}
          placeholder="Phone Number"
          type="text"
          value=""
        />
      </div>
      <div
        className="form-group row"
      >
        <label
          className="form-label"
          htmlFor="formPlaintextEmail"
        >
          Beginning Account Balance
        </label>
        <input
          className="form-control"
          id="formPlaintextEmail"
          name="name"
          onChange={[Function]}
          placeholder="Beginning Account Balance"
          type="text"
          value={0}
        />
      </div>
      <button
        className="d-inline-block btn btn-primary"
        disabled={false}
        to="http://localhost:3000/"
        type="submit"
      >
        Submit Application
      </button>
      <a
        href="/ThankYou"
        onClick={[Function]}
      />
    </form>
  </div>
</div>
`;
